### PR TITLE
Support filtering by has_many associations, without having to specify a custom filter

### PR DESCRIPTION
**Pull request by kuahyeow**

### CODE REVIEW

File: test/table_test.rb [hunk 1]
```diff
@@ -15,4 +15,30 @@
       table.candidate_parent_classes.must_equal []
     end
   end
+
+  describe "#" do
```
Comment (kuahyeow) [inline test/table_test.rb]: ooops

File: test/table_test.rb [hunk 2]
```diff
@@ -15,4 +15,30 @@
...
+    it "emits the tag_id condition for belongs_to" do
+      table = PartialKs::Table.new(BlogPost)
+      table.relation_for_association(User).to_sql.must_equal "SELECT \"blog_posts\".* FROM \"blog_posts\" WHERE \"blog_posts\".\"user_id\" = 0"
+    end
+
+    it "emits the subquery for has_many" do
+      table = PartialKs::Table.new(BlogPost)
+      post_tag_id = 111
+
+      PostTag.stub :pluck, [post_tag_id] do    # TODO verify the pluck
```
Comment (kuahyeow) [inline test/table_test.rb]: Ahem, TODO

File: test/table_test.rb [hunk 3]
```diff
@@ -15,4 +15,30 @@
...
+        table.relation_for_association(PostTag).to_sql.must_equal("SELECT \"blog_posts\".* FROM \"blog_posts\" WHERE \"blog_posts\".\"id\" IN (0, #{post_tag_id})")
+      end
+    end
+
+    describe "table with different :foreign_key" do
+      let(:table) { PartialKs::Table.new(OldTag) }
+
+      it "uses the foreign key that's present in the table" do
+        table.relation_for_association(OldEntry).to_sql.must_equal "SELECT \"old_tags\".* FROM \"old_tags\" WHERE \"old_tags\".\"blog_post_id\" = 0"
```
Comment (kuahyeow) [inline test/table_test.rb]: These SELECT queries feel quite brittle. No good ideas as to alternatives currently but it does lock one into the vagaries of how ActiveRecord emits SQL statements. Is a small hassle to update tests, is all

File: test/table_test.rb [hunk 4]
```diff
@@ -15,4 +15,39 @@
...
+    let(:table) { PartialKs::Table.new(BlogPost) }
+
+    it "emits the correct foreign key for the table for belongs_to" do
+      table.relation_for_association(User).must_equal BlogPost.where(:user_id => 0)
+    end
+
+    it "use results from pluck for belongs_to" do
+      user_ids = [1,2,3]
+      User.stub :pluck, user_ids do
+        table.relation_for_association(User).must_equal BlogPost.where(:user_id => [0] + user_ids)
```
Comment (dylanmckay) [inline test/table_test.rb]: Nit: you use hash rockets here but the newer syntax further up the diff

File: lib/partial_ks/table.rb [hunk 5]
```diff
@@ -27,6 +27,20 @@ def parent_tables
       belongs_to_reflections.map(&:table_name)
     end
 
+    def relation_for_association(klass)
```
Comment (dylanmckay) [inline lib/partial_ks/table.rb]: `relation_for_association` strongly implies it takes an `ActiveRecord::Relation`, but it actually takes a model and deduces the association from it.

Perhaps `relation_for_associated_model` could be more suitable?

**Comment by dylanmckay** [comment]
Nice change!

I've left two minor suggestions, otherwise looks good to me!

**Comment by kuahyeow** [comment]
Thanks ! Hmm, I can't see those suggestions @dylanmckay

**Comment by dylanmckay** [comment]
Ah, I'm not used to GitHub's review feature. Should be up now.

**Review by dylanmckay — CHANGES_REQUESTED** [review]

File: lib/partial_ks/table.rb (hunk 5 above)
Comment (kuahyeow) [reply lib/partial_ks/table.rb]: That's a perfect name! will change.

File: test/table_test.rb (hunk 4 above)
Comment (kuahyeow) [reply test/table_test.rb]: Hmm, looks like I use hash rockets everywhere prior to this change, but where I use `:` it's used for passing in keyword arguments. I would suggest fixing any consistency if any, later ?

Comment (dylanmckay) [reply test/table_test.rb]: Sounds good to me

**Review by dylanmckay — APPROVED** [review]
Nice work!

**Pull request merged**